sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/EventGridExtensionConfigProvider.cs

```diff
@@ -123,91 +123,48 @@ private async Task<HttpResponseMessage> ProcessAsync(HttpRequestMessage req)
+        {
+            List<Task<FunctionResult>> executions = new List<Task<FunctionResult>>();
 
-                // Single Dispatch
-                if (_listeners[functionName].SingleDispatch)
-                {
-                    foreach (var ev in events)
-                    {
-                        // assume each event is a JObject
-                        TriggeredFunctionData triggerData = new TriggeredFunctionData
-                        {
-                            TriggerValue = ev
-                        };
-                        executions.Add(_listeners[functionName].Executor.TryExecuteAsync(triggerData, CancellationToken.None));
-                    }
-                    await Task.WhenAll(executions).ConfigureAwait(false);
-                }
-                // Batch Dispatch
-                else
+            // Single Dispatch
+            if (_listeners[functionName].SingleDispatch)
+            {
+                foreach (var ev in events)
                 {
+                    // assume each event is a JObject
                     TriggeredFunctionData triggerData = new TriggeredFunctionData
                     {
-                        TriggerValue = events
+                        TriggerValue = ev
                     };
                     executions.Add(_listeners[functionName].Executor.TryExecuteAsync(triggerData, CancellationToken.None));
                 }
-
-                // FIXME without internal queuing, we are going to process all events in parallel
-                // and return 500 if there's at least one failure...which will cause EventGrid to resend the entire payload
-                foreach (var execution in executions)
-                {
-                    if (!execution.Result.Succeeded)
-                    {
-                        return new HttpResponseMessage(HttpStatusCode.InternalServerError) { Content = new StringContent(execution.Result.Exception.Message) };
-                    }
-                }
-
-                return new HttpResponseMessage(HttpStatusCode.Accepted);
+                await Task.WhenAll(executions).ConfigureAwait(false);
             }
-            else if (String.Equals(eventTypeHeader, "Unsubscribe", StringComparison.OrdinalIgnoreCase))
+            // Batch Dispatch
+            else
             {
-                // TODO disable function?
-                return new HttpResponseMessage(HttpStatusCode.Accepted);
+                TriggeredFunctionData triggerData = new TriggeredFunctionData
+                {
+                    TriggerValue = events
+                };
+                executions.Add(_listeners[functionName].Executor.TryExecuteAsync(triggerData, CancellationToken.None));
             }
 
-            return new HttpResponseMessage(HttpStatusCode.BadRequest);
+            // FIXME without internal queuing, we are going to process all events in parallel
```

Comment:
Should we file an issue?

Reply:
This will require some kind of internal queue(like for blob storage trigger) and redesign of EventGrid extension. Added an issue: [https://github.com/Azure/azure-sdk-for-net/issues/17756](https://github.com/Azure/azure-sdk-for-net/issues/17756)